src/server.c

```diff
@@ -1929,27 +1929,26 @@ void updateCachedTime(int update_daylight_info) {
 }
 
 void checkChildrenDone(void) {
-    int statloc;
+    int statloc = 0;
     pid_t pid;
 
-    if ((pid = wait3(&statloc,WNOHANG,NULL)) != 0) {
-        int exitcode = WEXITSTATUS(statloc);
+    if ((pid = waitpid(-1, &statloc, WNOHANG)) != 0) {
+        int exitcode = WIFEXITED(statloc) ? WEXITSTATUS(statloc) : -1;
```

Comment:
Why set the exitcode to 0 when WIFEXITED returns 0(process exit success)?
This may cause the `exitcode == 0` judgment to not even be valid later.

Reply:
Sorry I don't quite understand the comment here. I am setting `exitcode` to `-1` here if `WIFEXITED` returns 0.

Reply:
Sorry, I misunderstood, and you were right.